api/src/shop/statistics.py

```diff
@@ -0,0 +1,123 @@
+import datetime as datetime_module
+from dataclasses import dataclass, field
+from datetime import date, datetime
+from typing import Any, List, Optional, cast
+
+import sqlalchemy
+from dataclasses_json import DataClassJsonMixin, config
+from service.db import db_session
+from service.util import date_to_str, format_datetime
+from sqlalchemy import ColumnElement, func, select
+
+from shop.models import Product, Transaction, TransactionContent
+
+
+@dataclass
+class SalesByDate(DataClassJsonMixin):
+    date: datetime_module.date = field(metadata=config(encoder=date_to_str))
+    amount: float
+    count: int
+
+
+@dataclass
+class SalesResponse(DataClassJsonMixin):
+    id: int
+    total_amount: float
+    total_count: int
+    by_date: List[SalesByDate]
+    start_time: Optional[datetime] = field(metadata=config(encoder=format_datetime))
+    end_time: Optional[datetime] = field(metadata=config(encoder=format_datetime))
+    amount_unit: str
+
+
+INCLUDE: ColumnElement[bool] = cast(ColumnElement[bool], cast(Any, True))
```

Comment:
`INCLUDE` might require a comment explaining it's usage...

_Or_ perhaps it would be easier to understand by renaming the variable. Perhaps `anything_matches` would be better (if I understand the usage correctly!)?

Then it would read as:

```python
.where(Transaction.created_at >= start if start is not None else anything_matches)
```

Reply:
Improved